test/components/collision_detection_test.dart

```diff
@@ -0,0 +1,540 @@
+import 'package:flame/components.dart';
+import 'package:flame/geometry.dart' as geometry;
+import 'package:flame/src/geometry/circle.dart';
+import 'package:flame/src/geometry/line_segment.dart';
+import 'package:flame/src/geometry/line.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('LineSegment.isPointOnSegment tests', () {
+    test('Can catch simple point', () {
+      final segment = LineSegment(
+        Vector2.all(0),
+        Vector2.all(1),
+      );
+      final point = Vector2.all(0.5);
+      assert(segment.containsPoint(point), "Point should be on segment");
+    });
+
+    test('Should not catch point outside of segment', () {
+      final segment = LineSegment(
+        Vector2.all(0),
+        Vector2.all(1),
+      );
+      final point = Vector2.all(3);
```

Comment:
for absolute completeness we can have a test checking a point on the extended line and one not, both outside the segment